extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iProcessor.java

```diff
@@ -65,12 +83,36 @@ public CapabilityBuildItem capability() {
         return new CapabilityBuildItem(Capability.CONTAINER_IMAGE_S2I);
     }
 
+    @BuildStep(onlyIf = { S2iBuild.class }, onlyIfNot = NativeBuild.class)
+    public void s2iPrepareJvmDockerBuild(S2iConfig s2iConfig,
+            OutputTargetBuildItem out,
+            BuildProducer<DecoratorBuildItem> decorator) {
+        if (s2iConfig.buildStrategy == BuildStrategy.docker) {
+            decorator.produce(new DecoratorBuildItem(new ApplyDockerfileToBuildConfigDecorator(null,
+                    findMainSourcesRoot(out.getOutputDirectory()).getValue().resolve(s2iConfig.jvmDockerfile))));
+            decorator.produce(new DecoratorBuildItem(new RemoveEnvVarDecorator(null, "JAVA_APP_JAR")));
+        }
+    }
+
+    @BuildStep(onlyIf = { S2iBuild.class, NativeBuild.class })
+    public void s2iPrepareNativeDockerBuild(S2iConfig s2iConfig,
+            OutputTargetBuildItem out,
+            BuildProducer<DecoratorBuildItem> decorator) {
+        if (s2iConfig.buildStrategy == BuildStrategy.docker) {
+            decorator.produce(new DecoratorBuildItem(new ApplyDockerfileToBuildConfigDecorator(null,
+                    findMainSourcesRoot(out.getOutputDirectory()).getValue().resolve(s2iConfig.nativeDockerfile))));
```

Comment:
We should probably move this some utility common to all container-image extensions - It doesn't need to be now of course

Reply:
This is already in the quarkus container image util module.